util/src/main/java/org/hyperledger/besu/util/Subscribers.java

```diff
@@ -101,17 +102,19 @@ public boolean unsubscribe(final long subscriberId) {
    * @param action the action to perform for each subscriber
    */
   public void forEach(final Consumer<T> action) {
-    subscribers
-        .values()
+    ImmutableSet.copyOf(
+            // we copy here to ensure that our callback, which may add another subscriber on this
+            // event, won't trigger that new additional subscriber
+            subscribers.values())
         .forEach(
             subscriber -> {
               try {
                 action.accept(subscriber);
-              } catch (Throwable throwable) {
+              } catch (Exception e) {
```

Comment:
changing this from throwable to exception _seems_ important - i.e. prior to this change, throwables would be eaten by the "suppress" - now they will always be raised (which maybe the intent, but wanted to confirm).

Reply:
The only relevant throwable suppressed are `Error`s, which generally speaking are non-recoverable (`OutOfMemoryError, `ThreadDeath`, `LinkageError`).  When those happen the VM is going down hard.

Reply:
@rain-on Are you alright with leaving it as is?

Reply:
👍  all good.